app/views/contributions/show.html.erb

```diff
@@ -7,8 +7,8 @@
     <p class="subtitle"><%= contribution.description %></p>
     <% if contribution.matched? %>
       <i class="far fa-check-circle has-text-primary is-size-5 has-text-weight-semibold"><span class="ml-1">Claimed</span></i>
-    <% else %>
-      <%= button_to "Claim This #{contribution.type}", new_contribution_claim_path(contribution), class: "button is-primary", method: :get %>
+    <% else  %>
+      <%= button_to "Claim This #{contribution.type}", new_contribution_claim_path(contribution), class: "button is-primary", disabled: not_claimable?(contribution), method: :get %>
```

Comment:
@h-m-m Is using the `not_claimable?` helper method like this more along the lines of what you were thinking?

I was discussing this with @leahriffell and we were thinking that we wanted to put the helper method in the contribution model instead that way we could do a unit test on it but I'm not sure how the current `contribution.rb` file works with the `TYPES = [Ask, Offer]`

Reply:
I definitely think this is an improvement. I checked with @exbinary, who thinks we're going to want to make further changes to the logic soon. He may have some further suggestions, but whatever that plan is, pulling the logic out like this will make it easier for us to get there

The only remaining question I have, aside from testing concerns, is if it's better to disable the button or hide it. Or if we disable it, should there be some other message to the user?

I don't know what's the right thing here from a UX perspective, so I'm curious what you think

Reply:
I would think that disabling the button _and_ adding an explanatory message would be a less confusing experience than the button disappearing completely. Maybe something like "An ask without an email address cannot be claimed", "An ask without an email address cannot be claimed in peer-to-peer mode", etc.

Reply:
I agree with Leah. I like the idea of disabling it with a message better than hiding it. When we were looking at this together last week there was a flash message that explained the situation but I don't see it anymore. Regardless, I think the message should be permanent and near the button for better UX.